Members only Feature completion

RNtaate opened the pull request:
### What does this PR do?
- [X] Creates a Post model.
- [X] Installs the devise gem.
- [X] Creates a User model using devise.
- [X] Creates association between the Post and User model.
- [X] Creates user authentication using devise.
- [X] Creates User authorization for accessing certain user posts.
- [X] Styles posts, devise and layouts view templates using bulma classes.

ershadul1 committed:
Add rails starter files and necessary gems

RNtaate committed:
Create Post model using scaffold

RNtaate committed:
Add root route

RNtaate committed:
Add textarea to form partial

RNtaate committed:
Update post_params method in post controller

RNtaate committed:
Add post in index.html.erb

RNtaate committed:
Install devise gem and generate devise views

RNtaate committed:
Create User model using devise

ershadul1 committed:
Add name and username input fields in sign up form

ershadul1 committed:
Add registrations controller

ershadul1 committed:
Update route for devise

ershadul1 committed:
Add associations between posts and users

ershadul1 committed:
Add authentication for posts

ershadul1 committed:
Add name in the index.html.erb

ershadul1 committed:
Update new and create method to use current_user

ershadul1 committed:
Add authorization functionality to update, destroy methods

ershadul1 committed:
Update index.html.erb for authorization

ershadul1 committed:
Update show.html.erb of post view

RNtaate committed:
Change application.css to scss file, delete scaffold.scss

RNtaate committed:
Add styling to is-medium class buttons

RNtaate committed:
Re-design sign up view template

RNtaate committed:
Re-design sign in view template using bulma classes

RNtaate committed:
Re-name login to sign in

ershadul1 committed:
Add user_status column in users table

ershadul1 committed:
Update the index method to show user status

ershadul1 committed:
Add sign up, sign in, sign out and new post links in application layout

ershadul1 committed:
Add feed partial and render it inside index

RNtaate committed:
Update index method to check authentication in posts controller

RNtaate committed:
Update styling for index.html.erb using bulma class

RNtaate committed:
Re-design form partial using bulma classes

RNtaate committed:
Re-design new.html.erb using bulma classes

RNtaate committed:
Re-design application layout using bulma classes

RNtaate committed:
Add styling to notifications bar

RNtaate committed:
Re-design edit.html.erb using bulma classes

RNtaate committed:
Add ordering to display of posts in index method

RNtaate committed:
Render form partial in feed partial

RNtaate committed:
Add missing end div tag

RNtaate committed:
Fix rubocop offenses and sytlelint errors

ershadul1 committed:
Update README.md

ershadul1 committed:
Modify gemfile to deploy on heroku

RNtaate committed:
Update README file with live demo link

RNtaate committed:
Update README.md

kubilaycaglayan commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,83 @@
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```
:bulb: In the `new` method, the conventional way to initialize a post is as follows:
`@post = Post.new`
Since you are already using `build` in the create method, we don't need it here.

kubilaycaglayan commented on app/views/layouts/application.html.erb:
```diff
@@ -0,0 +1,72 @@
+  </head>
+
+  <body>
+ <%if flash[:notice]%>
+      <div class="notification is-primary global-notification">
+        <p class="notice"><%= notice %></p>
+      </div>
+    <%end%>
+
+    <% if flash[:alert]%>
```
:bulb: Move all the logic out of the views.
This `if block` and all the others.

kubilaycaglayan commented on app/views/posts/_feed.html.erb:
```diff
@@ -0,0 +1,43 @@
+<div class="column is-half">
+  <% if user_signed_in? %>
+  <article class="media box">
+    <div class="media-content">
+      <small>*Create more than 2 posts to get active user status*</small>
+      <%= render 'posts/form'%>
```
:bulb: Move all the logic out of the views.
This if block and all the others.

kubilaycaglayan reviewed (changes requested):
Hello Team 👋
Status: Changes Required :recycle: 

Great job so far! Wonderful styling 👏👋
Amazing readme 🗒️🎉

Let's cover some details together to make your project better ⏬

### Readme

- You have a wonderful readme file 🧡 Let's add `bundle` step to the "Steps To Follow" right before the 3rd step 🙏

### Routes

- [x] It creates the application routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

💡 I can see the extra work you have put, but please, let's follow the requirements.
Scaffolding is a great tool, but we should have a good understanding of these details.

### Controllers

- [x] it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

💡 Please check the inline comments ⏬

### Views

- [x] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.

[HINT]: Make helper methods for your own purpose in the `helpers` folder. You can use them in your views. You can call those helper methods in the views. You can pass parameters into them. And you can even return HTML partials.

💡 Please check the inline comments ⏬
💡💡 https://www.rubyguides.com/2020/01/rails-helpers/ ➡️ Here, follow the "Writing your own helper methods" and "How to Use Helpers From Controllers" titles should work for you.

### That's it! After doing the changes, you can request another review 😊

Thanks for your efforts and time...
With all my respect 🙏

RNtaate committed:
Add bundle install step in steps to follow section

RNtaate committed:
Update  the routes for posts for only index, create, new

RNtaate committed:
Update new method in posts controller

RNtaate committed:
Add show_only_to_user method

RNtaate committed:
Add post_name, post_user_status methods

RNtaate committed:
Create controls_in partial

RNtaate committed:
Create controls_out partial

RNtaate committed:
Remove logic from views

RNtaate committed:
Add styling to notice and alert flash messages

RNtaate committed:
Fix rubocop offenses

RNtaate commented:
## Changes made
- [X] Added `bundle` instruction in the steps to follow section in the README.
- [X] Updated the routes for posts using `only: [:new, :create, :index]`.
- [X] Initialized an empty post in the new method using `Post.new` instead of `build` in the posts controller
- [X] Used helper methods instead of using logic in view files.

trekab reviewed (approved):
### Approved! :tada:
Congratulations @RNtaate and @ershadul1! You have met the project requirements and you have done a great job implementing the feedback from the last review. Keep up the good work! :+1: 

_Happy coding!_ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)
